lib/interactive_viewer.ml

```diff
@@ -25,10 +29,25 @@ let current_hunks patches =
       Ui.vcat @@ List.map (fun p -> W.string @@ string_of_hunk p) p.Patch.hunks
   | None -> W.string "No operation"
 
+(** helper function to navigate between patches via `Next or `Prev *)
+let navigate direction zipper_of_patches =
+  let z = Lwd.peek zipper_of_patches in
+  match direction with
+  | `Next ->
+      if List.length z.after > 0 then begin
```

Comment:
Hi @panglesdm, do you mind clarifying what you mean by managing the `index_v` value? As of now, the current implementation for `next` and `prev` does not traverse any further once we reach the edge of the zipper. I believe we can customise this branch of the match statement to raise exceptions or log something to the user. In this sense, doesn't it make `index_v` obsolete? Otherwise I am misunderstanding something.

```
  let next z =
    match z.after with
    | [] -> z
    | x :: xs -> { before = z.focus :: z.before; focus = x; after = xs }

  let prev z =
    match z.before with
    | [] -> z
    | x :: xs -> { before = xs; focus = x; after = z.focus :: z.after }
```

P.S.: I pushed my current progress to the PR if you would like to take a look.

Reply:
Yes, sure!

So, the `index_v` value contains an integer, which is updated to contain the index of the operation we are currently looking at.
Previously, this was used to get the current operation (with `List.nth_opt`). This was changed by your PR, and you now only use it to check bounds to avoid going "previous" or "next" when you are at the bounds.
However:
- It is in general good to avoid having to check bounds for each call of `Zipper.prev` or `Zipper.next`,
- In this case, it is not really useful to check the bounds: in your last version, you do not modify the zipper in the "out of bounds" situation!

We could completely get rid of the `index` value, since as we do not have to check the bounds, we are not using this value.
For another PR, we could want to have this value to add "Operation x out of X" in the top bar. When this happen, we will want to have Zipper itself manage the `index` value, instead of updating it at each call of `Zipper.next/prev`.

I hope it is clearer, do let me know otherwise!

Reply:
Resolved